core/storage/translation/gae_models.py

```diff
@@ -0,0 +1,147 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for machine translation."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+import utils
+
+(base_models,) = models.Registry.import_models(
+    [models.NAMES.base_model])
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class MachineTranslatedTextModel(base_models.BaseModel):
+    """Model for storing machine translations. Model instances have a key
+    generated from the source and target language codes, followed by a sha-1
+    hash of the origin text formated as follows:
+
+        [source_language_code]:[target_language_code]:[hashed_origin_text]
```

Comment:
In many other places we use either `.` or `-` to divide different parts of the ID, why is `:` used here?

Reply:
I originally was using the `|` character but replaced it with `:` to make the model id regex more readable. I prefer to avoid using `-` because it sometimes appears in language codes, and `.` is a special character in regular expressions as well.
Other than readability though, there's no reason not to use `.`. Do you suggest changing it?

Reply:
Yep, lets use `.` since it used accross the codebase and you can always escape it in the regexes.

Reply:
Fixed.